src/utils/extract.js

```diff
@@ -80,5 +80,27 @@ export const extract = {
         }
 
         return '#' + redHex + greenHex + blueHex;
+    },
+    rgbToHSB: (red, green, blue) => {
+        let redQuotient = red/255;
+        let greenQuotient = green/255;
+        let blueQuotient = blue/255;
+        let minRGB = Math.min(redQuotient, Math.min(greenQuotient, blueQuotient));
+        let maxRGB = Math.max(redQuotient, Math.max(greenQuotient, blueQuotient));
+
+        // detects if color is white, gray, or black
+        if (minRGB===maxRGB) {
+            return [0,0,minRGB];
+        }
+        else {
+            let d = (redQuotient === minRGB) ? greenQuotient - blueQuotient : ((blueQuotient === minRGB) ? redQuotient - greenQuotient : blueQuotient - redQuotient);
```

Comment:
what is d? what is h? Could we name these more specifically?

Reply:
The formulas are derived from this resource: http://www.javascripter.net/faq/rgb2hsv.htm but the writer does not define what d and h represent within the formula. Further research on the formula leaves this writer's formula ambiguous still. Open to suggestion

Reply:
I appreciate that the code solves the problem. I think it would be nice to know what the variables mean in case we would ever need to use them for solutions in the future.

Reply:
We can move past this. I'll approve if you just address the first issue.